src/quick-lint-js/fe/parse-class.cpp

```diff
@@ -1183,7 +1191,10 @@ void parser::parse_and_visit_typescript_interface(
   case token_type::kw_assert:
   case token_type::kw_asserts:
   case token_type::kw_async:
-  case token_type::kw_constructor:
+  case token_type::kw_constructor: {
+    constructor_guard g = this->enter_constructor();
+    [[fallthrough]];
```

Comment:
Must fix: This cannot work. The `constructor_guard` is destructed before the fallthrough, and this code also runs if the token is `kw_async` (for example), not just for `kw_constructor`. I think you didn't test this code path.